src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -41,4 +41,9 @@ public ShoppingCart getByUser(User user) {
     public void registerNewShoppingCart(User user) {
         shoppingCartDao.add(new ShoppingCart(user));
     }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        ticketDao.deleteFromUserShoppingCart(shoppingCart.getUser());
```

Comment:
can we use shoppingCart.getByUser() and update() Here?

Reply:
Yes, but in my case it also lead to many delete queries (one for each item in shopping cart)

Reply:
And I think the answer may be "No", because shoppingCart.getByUser() will be a redundant query. We can use something like this:
        shoppingCart.getTickets().clear();
        update(shoppingCart);
And shoppingCart.getTickets() is just a getter.

Reply:
you right, meant to pass shoppingCart from shoppingCart.getByUser() to this method)
" I used something like this:
shoppingCart.getTickets().clear();
update(shoppingCart);" - exactly what I've meant))